### ISSUE TITLE
Colon in environment name results in misconfigured PATH

### ISSUE THREAD

**Issue by Machione** [conda/conda#7716]
Installed a fresh version of Miniconda in `/home/ryan/Conda` and created a new environment with a colon (:) in its name. Activating my new environment and running Python results in the base Python installation being called (`/home/ryan/Conda/bin/python`) rather than Python installed in my new environment (`/home/ryan/Conda/envs/.../bin/python`), meaning that any installed Python packages are not importable. Perhaps Conda should not allow colons to be present in environment names? Or alternatively, this limitation should be made clear in documentation and a meaningful warning printed to the user when they try to do so.

## Current Behavior
Environment names containing colons result in the PATH environment variable being incorrectly set upon activation. This is due to colons being used as a separator between paths in PATH. Escaping of colons in PATH is not possible, and therefore no path in PATH should contain a colon.
<http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html#tag_08_03>
<https://stackoverflow.com/questions/14661373/how-to-escape-colon-in-path-on-unix>

### Steps to Reproduce
```
$ conda create -n my:env python
```
Completes without error or suggestion that the user may want to reconsider their choice of environment name. All expected packages installed in `/home/ryan/Conda/envs/my:env`.
```
$ source activate my:env
```
Again, completes without complaining. Confirmed successfully working within my:env due to the (my:env) prefix on the prompt.
```
$ which python
```
Returns `/home/ryan/Conda/bin/python`, not the Python installed in the newly created environment which is located in `/home/ryan/Conda/envs/my:env/bin/`.
```
$ echo $PATH
```
Reveals the true problem; namely that this environment variable is set to `/home/ryan/Conda/envs/my:env/bin:/home/ryan/Conda/bin:...`. Notice the additional colon that separates what should be a single path `/home/ryan/Conda/envs/my:env/bin` into `/home/ryan/Conda/envs/my` and `env/bin`, which are illegitimate.

## Expected Behavior
An error thrown to the user upon the attempted creation of an environment containing a colon in its name. The error should describe that such names are invalid.

## Environment Information
<details><summary><code>`conda info`</code></summary><p>

```
     active environment : my:env
    active env location : /home/ryan/Conda/envs/my:env
            shell level : 1
       user config file : /home/ryan/.condarc
 populated config files : 
          conda version : 4.5.4
    conda-build version : not installed
         python version : 3.6.5.final.0
       base environment : /home/ryan/Conda  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/linux-64
                          https://repo.anaconda.com/pkgs/pro/noarch
          package cache : /home/ryan/Conda/pkgs
                          /home/ryan/.conda/pkgs
       envs directories : /home/ryan/Conda/envs
                          /home/ryan/.conda/envs
               platform : linux-64
             user-agent : conda/4.5.4 requests/2.18.4 CPython/3.6.5 Linux/4.18.4-arch1-1-ARCH arch/ glibc/2.28
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```
</p></details>

<details><summary><code>`conda config --show-sources`</code></summary><p>

```

```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
# packages in environment at /home/ryan/Conda/envs/my:env:
#
# Name                    Version                   Build  Channel
ca-certificates           2018.03.07                    0    defaults
certifi                   2018.8.24                py37_1    defaults
libedit                   3.1.20170329         h6b74fdf_2    defaults
libffi                    3.2.1                hd88cf55_4    defaults
libgcc-ng                 8.2.0                hdf63c60_1    defaults
libstdcxx-ng              8.2.0                hdf63c60_1    defaults
ncurses                   6.1                  hf484d3e_0    defaults
openssl                   1.0.2p               h14c3975_0    defaults
pip                       10.0.1                   py37_0    defaults
python                    3.7.0                hc3d631a_0    defaults
readline                  7.0                  h7b6447c_5    defaults
setuptools                40.2.0                   py37_0    defaults
sqlite                    3.24.0               h84994c4_0    defaults
tk                        8.6.8                hbc83047_0    defaults
wheel                     0.31.1                   py37_0    defaults
xz                        5.2.4                h14c3975_4    defaults
zlib                      1.2.11               ha838bed_2    defaults
```
</p></details>

**Comment by jachung2** [comment]
This was fixed in PR #7801

**Comment by jjhelmus** [comment]
Great job tracking this down @jachung2
